### DISCUSSION TITLE
Able to not set a relay for Light Area

### DISCUSSION THREAD

**Discussion by MakoShark2** [theyosh/TerrariumPI#935] [Ideas]
This one should be easy I think.

Here's my situation, I'm using a Jungle Dawn light setup which has it's own incorporated timer and from my knowledge is not supported by TerrariumPi. Which is fine, I could just create a Light area with the same timer schedule in TerrariumPi, only the area requests that a relay is picked up and without that I cannot create the area...
So If I'm correct and there is any other implication with it, maybe just making the relay option not required or adding a "None" or "Ignored" option would do.

Thanks.

Daniel Santos.

**Comment by theyosh** [comment]
An area must always have an relay. That is just mandatory. But you could just create a script relay that does nothing: https://theyosh.github.io/TerrariumPI/hardware/relay/script-relay/

And use that script relay in your area.

Example code: https://github.com/theyosh/TerrariumPI/blob/main/contrib/external_switch.py

**Comment by MakoShark2** [comment]
Looks like what I need. I was not aware of this... Thanks.

**Comment by MakoShark2** [comment]
Managed to make the script work.

For the readout I'm returning an interpolation of the relay value to mimic the lights which are custom dimmable.
But it's showing only as on or off (100.0 or 0.0).

Here's the code I'm using for the readout:

```
if len(sys.argv) == 1:
    # Call your readout or current state code here....

    # get time
    now = datetime.now()
    t = now.time().hour + (now.time().minute/60)

    # known points for lights schedules
    x_known = numpy.array([8, 11, 12.5, 15, 18, 20, 21, 22, 23, 23.5])
    y_known = numpy.array([0, 90, 100, 100, 80, 50, 5, 2, 1, 0])

    print(round(numpy.interp(t, x_known, y_known),2))
```

Which work as expected:

```
z06@GREMLIN:/opt/terrariumpi/scripts $ date
Fri 16 Aug 10:38:20 WEST 2024
z06@GREMLIN:/opt/terrariumpi/scripts $ sudo python dummy_lights_relay.py
79.0
z06@GREMLIN:/opt/terrariumpi/scripts $
```

And what's showing in the log:

```
2024-08-16 10:39:08,584 - INFO    - terrariumRelay        - Updated relay script relay named 'Lights' at address '/TerrariumPI/scripts/dummy_lights_relay.py' with new value 100.00 in 1.25 seconds.
```

This way wattage calculation migth be not that accurate but not a problem really.

Still maybe something I'm doing wrong?

**Reply by theyosh** [reply]
I should just return the value of "-1". As this will instruct TP not to use the readout and use the value that is set.

**Reply by theyosh** [reply]
And you should use `script dimmer` and not `script relay`

**Comment by MakoShark2** [comment]
Thanks, the script dimmer did the trick!

**Reply by MakoShark2** [reply]
So I have setup two relays like this:

#1 script dimmer to mimic my autonomous led bar, which returns different readouts along the day,
the script will provide a fading in (0 to 100%) readout from 8:00 to 12:30, then 100% from 12:30 to 15:00, and fading out (100 to 0%) from 15:00 to 23:30

#2 script relay (readout -1) , which is set as Main Lights.

Then I create the following areas:

#A area named "Day Lights" in timer mode (8:00 to 23:30) attached to #1 
#B name "Main Lights" in timer mode (8:00 to 20:00) attached to #2 and depends on #1 being on

For this scenario what I'm seeing is #B being switched on only when #1 is at 100%. (12:30 to 15:00).

I was expecting #2 (through #B) to be switched on as long as #1 > 0% but I might misinterpreted something.

**Reply by theyosh** [reply]
> I was expecting https://github.com/theyosh/TerrariumPI/issues/2 (through #B) to be switched on as long as https://github.com/theyosh/TerrariumPI/issues/1 > 0% but I might misinterpreted something.

No, the depends on is based on alarm state of the other area: https://theyosh.github.io/TerrariumPI/setup/#other-areas read below the image.

Also, when set an area as Main lights, you should have picked the type 'Lights' and that does not show the depending on option.

But for now, do not use the option depending on. If you have booth set on timer mode with the correct values, they will work in 'sync'

**Reply by MakoShark2** [reply]
Ah yes... Now I get it from the example... Makes sense, triggered by the alarm state.

I'll set the areas as you said, timers both depending only on the time of the day.

Thank you!
